Windows GA failing with compiler warning in drreg.c

A compiler update on the Windows GA has caused all our PR's (#7736, #7737, #7719, #7738) tests to fail with this warning:

```
D:\a\dynamorio\dynamorio\ext\drreg\drreg.c(423): error C2220: the following warning is treated as an error
D:\a\dynamorio\dynamorio\ext\drreg\drreg.c(423): warning C4319: '~': zero extending 'uint' to 'ptr_uint_t' of greater size
```

The fix is inside #7736 but it would be best to commit separately.

--- a/ext/drreg/drreg.c
+++ b/ext/drreg/drreg.c
@@ -408,7 +408,7 @@ drreg_event_bb_analysis(void *drcontext, void *tag, instrlist_t *bb, bool for_tr
         if (xfer)
             aflags_cur = EFLAGS_READ_ARITH; /* assume flags are read before written */
         else {
-            uint aflags_read, aflags_w2r;
+            ptr_uint_t aflags_read, aflags_w2r;
             if (index == 0)
                 aflags_cur = EFLAGS_READ_ARITH; /* assume flags are read before written */
             else {

